### PR TITLE
Upgrade sample app to rails 7

### PR DESCRIPTION
**Pull request by alextwoods**
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

File: sample_app/Gemfile [hunk 1]
```diff
@@ -7,7 +7,11 @@ gem 'aws-sdk-rails', path: '../'
 # gem 'aws-sessionstore-dynamodb', path: '../../aws-sessionstore-dynamodb-ruby'
 
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
-gem 'rails' #, path: '../../rails'
+gem 'rails', '~> 7.0'
```
Comment (mullermp) [inline sample_app/Gemfile:10]: I prefer to keep the comment (for development)

**Review by mullermp — APPROVED** [review]
Assuming most of these are "generated" changes from rails new?

File: sample_app/Gemfile (hunk 1 above)
Comment (alextwoods) [reply sample_app/Gemfile:10]: These are mostly changes from running a `rails app:upgrade` - we could instead create a new app (i think in particular rails 7 does not use webpack by default) and add our customizations in instead so we get a cleaner rails 7 app.  Thoughts?

Comment (mullermp) [reply sample_app/Gemfile:10]: I didn't know about that command. I'm sure it's fine.

**Pull request merged**